src/adams_methods.jl

```diff
@@ -0,0 +1,636 @@
+                    # not depend on dy_np1 or y_np1, so no update necessary
+
+                    # We correct the computation of y_np1 using y_np1
+                    # as part of our prediction Harier equation 5.7
+
+                    for i=1:dof
+                        y_np1[i] += dt*g[steporder+1]*ϕ[3,steporder+1][i]
+                    end
+
+                    # (E)valuate estimate for next derivative again
+                    dy_np1=F(t_np1,y_np1)
+                end
+            end
+            #---################################################################
+            # Error Check for step size `dt` and Variable next step selection:
+            #     Check error to determine if step size should be accepted:
+            #     ->if yes, compute `next_dt` for next step
+            #     ->if no, use this `next_dt`` for next attempt at this step
+            #---################################################################
+
+            #Don't allow increases of steps until n >= steporder + 2
+            if n==1
+                #TODO: adaptive step selection for n<steporder +2
+                successful_step = true
+                next_dt = dt/4
+            elseif n<steporder+2
+                for i = 1:dof
+                    yerr_k[i] = dt*(g[steporder+1] - g[steporder])*ϕ[3,steporder+1][i]
+                end
+                err_k, next_dt,timeout_VS = stepsize_hw92!(dt, tdir, y[n], y_np1, yerr_k, steporder,
+                                timeout_VS, dof, abstol, reltol, maxstep, norm)
+                if err_k <= 1
+                    successful_step = true
+                else
+                    successful_step = false
+                    dt = .5*next_dt
+                end
+            else
+                # We calculate local truncation error, LE_{k+1}; the difference
+                # between implicit results for order k and k+1.
+
+                # We need ϕ_{k+1}(n+1) and g_{k+1}(n);the latter has already
+                # been calculated above, the former still needs calculation
+                ϕ_coefs!(ϕ, ϕstar,dy_np1,steporder+2)
+
+                # Calculate ||LE_{k+1}|| and next stepsize
+                for i = 1: dof
+                    yerr_kp1[i] = dt*(g[steporder+2] - g[steporder+1])*ϕ[3,steporder+2][i]
+                end
+                err_kp1, next_dt, timeout_VS = stepsize_hw92!(dt,tdir,y[n],y_np1,
+                                            yerr_kp1, steporder, timeout_VS,dof,
+                                            abstol, reltol, maxstep, norm)
+
+                # Harier equation 7.4: If ||LE_{k+1}|| =< 1
+                # then step size is accepted and use next_dt for next step
+                # Else, then step size is rejected, reattempt with fac*next_dt
+                if err_kp1 <= 1
+                    successful_step = true
+                else
+                    successful_step = false
+                    dt = .75*next_dt # make sure fac < 1
```

Comment:
Yes, I made the switch, and, the choice of `fac` does change the results slightly, so it is a question of if we want to give the user this the ability to easily change this. However, to fit conventions, I'll remove the keyword option.